pkg/server/server.go

```diff
@@ -69,8 +69,8 @@ type Callbacks interface {
 }
 
 // NewServer creates handlers from a config watcher and callbacks.
-func NewServer(config cache.Cache, callbacks Callbacks) Server {
-	return &server{cache: config, callbacks: callbacks}
+func NewServer(config cache.Cache, callbacks Callbacks, done chan bool) Server {
+	return &server{cache: config, callbacks: callbacks, done: done}
```

Comment:
This looks fine, but I think it's more idiomatic to use `context.Context` and then select on `context.Done()`. WDYT? See https://golang.org/pkg/context/#Context

Reply:
Thank you for the suggestion. That completely makes sense.
I have corrected the usage now. LMK if you have more suggestions.